### PR TITLE
Fix broken SVG rendering in Firefox

### LINKED ISSUE #4266: Error in toggling flip book in Firefox
## Expected behavior

The user should be able to toggle flip book mode after adding annotations to the subject.

## Current behavior

See below for JS error console output. The error happens after clicking toggle flip book mode but before the subject images load.

It stops at this state:

screen shot 2018-01-08 at 11 34 28

## Steps to replicate

- Add a mark to (e.g.) a [Backyard Worlds subject](https://www.zooniverse.org/projects/marckuchner/backyard-worlds-planet-9/classify) with Firefox
- Click Toggle flipbook mode 

## Additional information

- **Operating system:** macOS 10.13.2 and Windows 10
- **Browser:** Firefox 57.0.4

```
Checking current user
vendor-f876843aa5e61bae2839.js:11:84968
Getting bearer token
vendor-f876843aa5e61bae2839.js:11:82177
Emptying upcoming subjects queue
main-9881bb6cc0b989b2825a.js:1:1455948
AppStatus: Received status data from https://static.zooniverse.org/zooniverse.org-status.txt.
main-9881bb6cc0b989b2825a.js:1:1076130
AppStatus: Nothing to report.
main-9881bb6cc0b989b2825a.js:1:1076599
Failed to get bearer token
vendor-f876843aa5e61bae2839.js:11:82580
No current user
vendor-f876843aa5e61bae2839.js:11:85140
Deployed back end is 4 commits behind master.
main-9881bb6cc0b989b2825a.js:1:1101403
Unexpected value NaN parsing y1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing y2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing x1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing x2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing y1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing y2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing x1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing x2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value NaN parsing r attribute.
vendor-f876843aa5e61bae2839.js:43:23469
TypeError: e is null
[Learn More]
main-9881bb6cc0b989b2825a.js:1:7869
Unexpected value -Infinity parsing y1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value -Infinity parsing y2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value -Infinity parsing x1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value -Infinity parsing x2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value Infinity parsing y1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value Infinity parsing y2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value Infinity parsing x1 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value Infinity parsing x2 attribute.
vendor-f876843aa5e61bae2839.js:43:23469
Unexpected value Infinity parsing r attribute.
vendor-f876843aa5e61bae2839.js:43:23469
TypeError: e is null
[Learn More]
main-9881bb6cc0b989b2825a.js:1:7869
TypeError: t is null
[Learn More]
vendor-f876843aa5e61bae2839.js:55:152748
```

### PR DESCRIPTION
**Pull request by eatyourgreens**
Staging branch URL: https://fix-4266.pfe-preview.zooniverse.org/

Fixes #4266 .

Describe your changes.
Tries to set sensible defaults for non-finite image scaling, and holds off rendering delete buttons and drag handles until the screen transformation matrix is defined. Also disables the flipbook buttons until the subject has loaded.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/classifier/annotation-renderer/svg.jsx [hunk 1]
```diff
@@ -30,8 +30,10 @@ export default class SVGRenderer extends React.Component {
 
   getScale() {
     const sizeRect = this.getSizeRect();
-    const horizontal = sizeRect ? sizeRect.width / this.props.naturalWidth : 0;
```
Comment (marten) [inline app/classifier/annotation-renderer/svg.jsx]: What does `0.001` mean? That sounds like it'll make everything tiny, wouldn't `1` be a more natural default scale factor?

Comment (eatyourgreens) [reply app/classifier/annotation-renderer/svg.jsx]: Setting a default of 1 means everything starts really big, then shrinks down to size once the image has loaded and the scale is set to, say, 0.5. This gives a noticeable jump in the size of the drawn markers, between the initial render and values being set, so I opted for a minimum value of `0.001` instead.

Comment (eatyourgreens) [reply app/classifier/annotation-renderer/svg.jsx]: That said, I am seeing an annoying pause in Firefox, sometimes, between the SVG image rendering and the drawn marks being rendered.

**Comment by eatyourgreens** [comment]
Here's a link for testing with Backyard Worlds: https://fix-4266.pfe-preview.zooniverse.org/projects/marckuchner/backyard-worlds-planet-9/classify?env=production

**Pull request merged**